### ISSUE TITLE
vue/multiline-html-element-content-newline option to ignore comments

### ISSUE THREAD

**Issue by dsl101** [vuejs/eslint-plugin-vue#2179]
**What rule do you want to change?**
vue/multiline-html-element-content-newline

**Does this change cause the rule to produce more or fewer warnings?**
Fewer if new option set

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option, or extension to the `ignores` option. Either a boolean option, default `false`, called something like `allowHTMLComments`, or a token such as `HTMLComments` added to the list of possible `ignores` values. Note I'm not sure about the latter, as my understanding is that ignores list is looking for the element names _before_ the subsequent content without newline (e.g. the `<div>`), where as we'd like to allow multiline for _any_ element, providing that the subsequent element is a comment.

**Please provide some example code that this change will affect:**
To enable this to be Good:

```vue
<div class="something">  <!-- or use class="other" -->
  <Component />
</div>
```

**What does the rule currently do for this code?**
Currently, it produces the error `vue/multiline-html-element-content-newline – Expected 1 line break after opening tab ('<div>'), but no line breaks found.`

**What will the rule do after it's changed?**
Produce no errors / warnings with same-line HTML comments

**Additional context**
This is a common pattern for our JS code, and we'd like to be able to use the same kind of inline comments within the template:
```js
const meaning = 42  // obviously
```

**Comment by FloEdelmann** [comment]
There is a partial implementation that might provide a good starting point for anyone wanting to take a stab at this:

- https://github.com/vuejs/eslint-plugin-vue/pull/2581#discussion_r1815248209
